app/models/accounting/transaction.rb

```diff
@@ -52,6 +57,14 @@ def self.find_or_create_from_qb_object(transaction_type:, qb_object:)
     transaction.save!(validate: false)
   end
 
+  def self.by_date
+    order(txn_date: :asc)
+  end
+
+  def self.previous(loan_id)
```

Comment:
`previous` needs to be an instance method, as it should be pointing to the instance before your current instance, and not just to the second most recent instance overall.